### ISSUE TITLE
Enum functional API is not fully supported

### ISSUE THREAD

**Issue by matangover** [microsoft/pyright#1134]
Enum functional API was partially implemented in https://github.com/microsoft/pyright/issues/134, but some constructs are not supported, e.g. using a sequence should also work:
```python
TEST_ENUM = Enum('TEST_ENUM', ['A', 'B', 'C', 'D'])
```

[Docs](https://docs.python.org/3/library/enum.html#functional-api)
> The second argument is the _source_ of enumeration member names. It can be a whitespace-separated string of names, a sequence of names, a sequence of 2-tuples with key/value pairs, or a mapping (e.g. dictionary) of names to values.

Seems that only a whitespace-separated string of names is supported, and the other cases give a wrong error message of 'Expected 1 positional argument'.

**Comment by erictraut** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/1127

**Comment by matangover** [comment]
Slrr
